js/src/quantity-selector.js

```diff
@@ -0,0 +1,153 @@
+ * Licensed under MIT (https://github.com/Orange-OpenSource/Orange-Boosted-Bootstrap/blob/main/LICENSE)
+ * --------------------------------------------------------------------------
+ */
+
+import { defineJQueryPlugin } from './util/index'
+import EventHandler from './dom/event-handler'
+import BaseComponent from './base-component'
+import SelectorEngine from './dom/selector-engine'
+
+/**
+ * Constants
+ */
+
+const NAME = 'quantityselector'
+const DATA_KEY = 'bs.quantityselector'
+const EVENT_KEY = `.${DATA_KEY}`
+const DATA_API_KEY = '.data-api'
+
+const EVENT_LOAD_DATA_API = `load${EVENT_KEY}${DATA_API_KEY}`
+const EVENT_CHANGE_DATA_API = `change${EVENT_KEY}${DATA_API_KEY}`
+const EVENT_CLICK_DATA_API = `click${EVENT_KEY}${DATA_API_KEY}`
+
+const SELECTOR_STEP_UP_BUTTON = '[data-bs-step="up"]'
+const SELECTOR_STEP_DOWN_BUTTON = '[data-bs-step="down"]'
+const SELECTOR_COUNTER_INPUT = '[data-bs-step="counter"]'
+const SELECTOR_INPUT_GROUP = '.input-group'
+
+/**
+ * Class definition
+ */
+
+class QuantitySelector extends BaseComponent {
+  // Getters
+  static get NAME() {
+    return NAME
+  }
+
+  // Public
+  ValueOnLoad(element) {
+    const counterInput = element.querySelector(SELECTOR_COUNTER_INPUT)
+    const btnUp = element.querySelector(SELECTOR_STEP_UP_BUTTON)
+    const btnDown = element.querySelector(SELECTOR_STEP_DOWN_BUTTON)
+
+    const min = counterInput.getAttribute('min')
+    const max = counterInput.getAttribute('max')
+    const step = Number(counterInput.getAttribute('step'))
+
+    if (Number(counterInput.value) - step < min) {
+      btnDown.setAttribute('disabled', '')
+    }
+
+    if (Number(counterInput.value) + step > max) {
+      btnUp.setAttribute('disabled', '')
+    }
+  }
+
+  // Static
+  static StepUp(event) {
+    const parent = event.target.closest(SELECTOR_INPUT_GROUP)
+    const counterInput = parent.querySelector(SELECTOR_COUNTER_INPUT)
+
+    const max = counterInput.getAttribute('max')
+    const step = Number(counterInput.getAttribute('step'))
+    const round = Number(counterInput.getAttribute('data-bs-round'))
+
+    const eventChange = new Event('change')
+
+    if (Number(counterInput.value) < max) {
+      counterInput.value = (Number(counterInput.value) + step).toFixed(round).toString()
+    }
+
+    counterInput.dispatchEvent(eventChange)
+  }
+
+  static StepDown(event) {
+    const parent = event.target.closest(SELECTOR_INPUT_GROUP)
+    const counterInput = parent.querySelector(SELECTOR_COUNTER_INPUT)
+
+    const min = counterInput.getAttribute('min')
+    const step = Number(counterInput.getAttribute('step'))
+    const round = Number(counterInput.getAttribute('data-bs-round'))
+
+    const eventChange = new Event('change')
+
+    if (Number(counterInput.value) > min) {
+      counterInput.value = (Number(counterInput.value) - step).toFixed(round).toString()
+    }
+
+    counterInput.dispatchEvent(eventChange)
+  }
+
+  static ValueChange(event) {
```

Comment:
Maybe this function should use round too ? (I don't know what was decided with accessibility about rewriting user input) if yes, you could simplify the Stepup and down functions !

Reply:
It's not the role of this function to calculate new values.

For me the only problem here is the name of the function. We may change it for something like 'CheckIfDisabledOnChange()'?